### PR TITLE
Updated README

### PR DESCRIPTION
**Pull request by KarimZaoui**

### CODE REVIEW

File: docs/pipeline.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+#### Releases
+
+
+####  Full YAML file example
+
+```yaml
+desc: My data pipeline description
+
+datasets: ['staging', 'data']
```
Comment (charlielewis) [inline docs/pipeline.md:177]: It is no longer necessary to list datasets in the YAML.

`pipeline.py` now reads the table_list instead.

File: docs/storage.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Python utility to perform task with the Google Cloud Storage API
+
+#### Example
+
+```python
+from pygyver.etl.storage import GCSExecutor
+
+# Login
+fb = GCSExecutor()
```
Comment (charlielewis) [inline docs/storage.md:13]: `fb` -> `gcs`

File: docs/pipeline.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+batches:
+  - desc: Create table in staging
+    tables:
+      - table_desc: Staging table for pipeline
+        create_table:
+          table_id: pipeline
+          dataset_id: staging
+          schema_path: pipeline/schema/staging_pipeline.json
+          file: pipeline/sql/staging_pipeline.json
+        pk: [pk]
```
Comment (charlielewis) [inline docs/pipeline.md:99]: At least one of the `pk` examples should include a comma separated list of fields to demonstrate this functionality fully.

File: docs/pipeline.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+  - desc: Create table in data
+    tables:
+      - table_desc: Final table for pipeline
+        create_table:
+          table_id: pipeline
+          dataset_id: data
+          schema_path: pipeline/schema/data_pipeline.json
+          file: pipeline/sql/data_pipeline.json
```
Comment (charlielewis) [inline docs/pipeline.md:198]: `pipeline/sql/data_pipeline.json` -> `pipeline/sql/data_pipeline.sql`

File: docs/pipeline.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+batches:
+
+  - desc: Create table in staging
+    tables:
+      - table_desc: Staging table for pipeline
+        create_table:
+          table_id: pipeline
+          dataset_id: staging
+          schema_path: pipeline/schema/staging_pipeline.json
+          file: pipeline/sql/staging_pipeline.json
```
Comment (charlielewis) [inline docs/pipeline.md:188]: `pipeline/sql/staging_pipeline.json` -> `pipeline/sql/staging_pipeline.sql`

File: docs/pipeline.md [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+```yaml
+batches:
+  - desc: Create table in staging
+    tables:
+      - table_desc: Staging table for pipeline
+        create_table:
+          table_id: pipeline
+          dataset_id: staging
+          schema_path: pipeline/schema/staging_pipeline.json
+          file: pipeline/sql/staging_pipeline.json
```
Comment (charlielewis) [inline docs/pipeline.md:98]: `pipeline/sql/staging_pipeline.json` -> `pipeline/sql/staging_pipeline.sql`

File: docs/pipeline.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
```
Comment (charlielewis) [inline docs/pipeline.md:117]: `pipeline/sql/staging_pipeline.json` -> `pipeline/sql/staging_pipeline.sql`

File: docs/pipeline.md [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+          mock_file: product/mock/staging_pipeline.sql
+          output_table_name: staging.pipeline
+  - desc: Create pipeline_2 in staging
+    tables:
+      - table_desc: Staging table for pipeline
+        create_table:
+          table_id: pipeline_2
+          dataset_id: staging
+          schema_path: pipeline/schema/staging_pipeline_2.json
+          file: pipeline/sql/staging_pipeline_2.json
```
Comment (charlielewis) [inline docs/pipeline.md:129]: `pipeline/sql/staging_pipeline_2.json` -> `pipeline/sql/staging_pipeline_2.sql`

File: docs/pipeline.md [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+```yaml
+batches:
+  - desc: Create table and pipeline_2 in staging
+    tables:
+      - table_desc: Staging table for pipeline
+        create_table:
+          table_id: pipeline
+          dataset_id: staging
+          schema_path: pipeline/schema/staging_pipeline.json
+          file: pipeline/sql/staging_pipeline.json
```
Comment (charlielewis) [inline docs/pipeline.md:148]: `pipeline/sql/staging_pipeline.json` -> `pipeline/sql/staging_pipeline.sql`

File: docs/pipeline.md [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+        pk: [pk]
+        mock_data: 
+          mock_file: product/mock/staging_pipeline.sql
+          output_table_name: staging.pipeline
+      - table_desc: Staging table for pipeline
+        create_table:
+          table_id: pipeline_2
+          dataset_id: staging
+          schema_path: pipeline/schema/staging_pipeline_2.json
+          file: pipeline/sql/staging_pipeline_2.json
```
Comment (charlielewis) [inline docs/pipeline.md:158]: `pipeline/sql/staging_pipeline_2.json` -> `pipeline/sql/staging_pipeline_2.sql`

File: docs/pipeline.md [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+    tables:
+      - table_desc: Staging table for pipeline
+        create_table:
+          table_id: pipeline
+          dataset_id: staging
+          schema_path: pipeline/schema/staging_pipeline.json
+          file: pipeline/sql/staging_pipeline.json
+        pk: [pk]
+        mock_data: 
+          mock_file: product/mock/staging_pipeline.sql
```
Comment (charlielewis) [inline docs/pipeline.md:101]: `product/mock/staging_pipeline.sql` -> `pipeline/mock/staging_pipeline.sql`

File: docs/pipeline.md (hunk 6 above)
Comment (charlielewis) [reply docs/pipeline.md:98]: Note this same error occurs multiple times...

File: docs/pipeline.md (hunk 11 above)
Comment (charlielewis) [inline docs/pipeline.md:101]: `product/mock/staging_pipeline.sql` -> `pipeline/mock/staging_pipeline.sql`

Comment (charlielewis) [reply docs/pipeline.md:101]: Note this same error occurs multiple times...

File: docs/dw.md [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+db = BigQueryExecutor()
+
+# Execute SQL query against BigQuery API
+sql = "SELECT * FROM data.sale_order"
+sale_order_data = db.execute_sql(
+    sql=sql
+)
+
+# Execute SQL file against BigQuery API
+sql_file = "src/extract_sale_order.sql"
```
Comment (charlielewis) [inline docs/dw.md:22]: This path is more likely to be something like:

`sql_file= "sale_order/sql/extract_sale_order.sql"`

File: docs/pipeline.md (hunk 1 above)
Comment (KarimZaoui) [reply docs/pipeline.md:205]: I know. I was wondering whether to include it in the doc or not as we currently have it referenced in several pipelines.

File: docs/dw.md (hunk 12 above)
Comment (KarimZaoui) [reply docs/dw.md:22]: the `dw.md` example is just very simple, I am not even referring to any folder structure.

**Pull request merged**